src/sentry/notifications/notifications/base.py

```diff
@@ -103,63 +106,43 @@ def get_message_description(self, recipient: Team | User) -> Any:
         context = getattr(self, "context", None)
         return context["text_description"] if context else None
 
-    def get_type(self) -> str:
-        raise NotImplementedError
-
     def get_unsubscribe_key(self) -> tuple[str, int, str | None] | None:
         return None
 
     def get_log_params(self, recipient: Team | User) -> Mapping[str, Any]:
+        """
+        Get the relevant data about a notification that should be passed to any
+        logging calls. Best practice is to call `super().get_log_params()` in
+        any overriding calls.
+        """
         return {
             "organization_id": self.organization.id,
             "actor_id": recipient.actor_id,
         }
 
-    def get_custom_analytics_params(self, recipient: Team | User) -> Mapping[str, Any]:
-        """
-        Returns a mapping of params used to record the event associated with self.analytics_event.
-        By default, use the log params.
-        """
-        return self.get_log_params(recipient)
-
     def get_message_actions(self, recipient: Team | User) -> Sequence[MessageAction]:
         return []
 
     def get_callback_data(self) -> Mapping[str, Any] | None:
         return None
 
-    @property
-    def analytics_instance(self) -> Any | None:
+    def record_notification_sent(self, recipient: Team | User, provider: ExternalProviders) -> None:
         """
-        Returns an instance for that can be used for analytics such as an organization or project
+        Record an analytics event for this notification. This is called by each
+        provider in `notify()`.
         """
-        return None
-
-    def record_analytics(self, event_name: str, *args: Any, **kwargs: Any) -> None:
-        analytics.record(event_name, *args, **kwargs)
-
-    def record_notification_sent(self, recipient: Team | User, provider: ExternalProviders) -> None:
```

Comment:
@mgaeta Can you pull out the analytics changes into a separate PR? I'd like review that separately.